### PR TITLE
feat(dcs): import DCS resources, datasources and docs

### PR DESCRIPTION
**Pull request by Zhukun-Huawei**
…document

### CODE REVIEW

**Review by chengxiangdong — APPROVED** [review]
/lgtm

File: flexibleengine/acceptance/acceptance.go [hunk 1]
```diff
@@ -154,4 +156,4 @@ func testAccPreCheckImsBackupId(t *testing.T) {
 	if OS_IMS_BACKUP_ID == "" {
 		t.Skip("OS_IMS_BACKUP_ID must be set for IMS whole image with CBR backup id")
 	}
-}
+}
```
Comment (chengxiangdong) [inline flexibleengine/acceptance/acceptance.go:159]: Please keep the last empty line.

File: flexibleengine/acceptance/common.go [hunk 2]
```diff
@@ -60,3 +60,62 @@ data "flexibleengine_images_image" "test" {
...
+  available_zones   = ["eu-west-0a", "eu-west-0b"]
+
+  save_days       = 1
+  backup_type     = "manual"
+  begin_at        = "00:00-01:00"
+  period_type     = "weekly"
+  backup_at       = [1]
+}
+`, testAccDcsV1Instance_network(rName), rName)
+}
```
Comment (chengxiangdong) [inline flexibleengine/acceptance/common.go:121]: Usually, we leave a empty line at the end of the file.

File: flexibleengine/provider.go [hunk 3]
```diff
@@ -522,6 +530,9 @@ func Provider() *schema.Provider {
 			"flexibleengine_rds_account":            rds.ResourceMysqlAccount(),
 			"flexibleengine_rds_database":           rds.ResourceMysqlDatabase(),
 			"flexibleengine_rds_database_privilege": rds.ResourceMysqlDatabasePrivilege(),
+
+			"flexibleengine_sdrs_protection_group":   sdrs.ResourceProtectionGroup(),
+
```
Comment (chengxiangdong) [inline flexibleengine/provider.go:535]: SDRS should create a new patch. Please remove these lines,

File: flexibleengine/acceptance/acceptance.go (hunk 1 above)
Comment (Zhukun-Huawei) [reply flexibleengine/acceptance/acceptance.go]: Done， I leave a last empty line.

File: flexibleengine/acceptance/common.go (hunk 2 above)
Comment (Zhukun-Huawei) [reply flexibleengine/acceptance/common.go]: Done， I leave a last empty line.

File: flexibleengine/provider.go (hunk 3 above)
Comment (Zhukun-Huawei) [reply flexibleengine/provider.go]: Done， I has removed these lines

**Comment by chengxiangdong** [comment]
The commit message "tmp" lacks clear meaning, pls refine it.

File: docs/data-sources/dcs_flavors.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+---
+subcategory: "Distributed Cache Service (DCS)"
+---
+
+# flexibleengine_dcs_flavors
```
Comment (chengxiangdong) [inline docs/data-sources/dcs_flavors.md:5]:
```suggestion
# flexibleengine_dcs_flavors_v1
```

File: flexibleengine/provider.go [hunk 5]
```diff
@@ -291,6 +292,9 @@ func Provider() *schema.Provider {
 
 			"flexibleengine_compute_servergroups": ecs.DataSourceComputeServerGroups(),
 
+			"flexibleengine_dcs_flavors":   dcs.DataSourceDcsFlavorsV2(),
+			"flexibleengine_dcs_instances": dcs.DataSourceDcsInstance(),
```
Comment (chengxiangdong) [inline flexibleengine/provider.go:296]:
```suggestion
			"flexibleengine_dcs_flavors_v1":   dcs.DataSourceDcsFlavorsV2(),
			"flexibleengine_dcs_instances_v1": dcs.DataSourceDcsInstance(),
```

File: flexibleengine/provider.go [hunk 6]
```diff
@@ -496,6 +500,9 @@ func Provider() *schema.Provider {
 			"flexibleengine_cse_microservice_engine":   cse.ResourceMicroserviceEngine(),
 			"flexibleengine_cse_microservice_instance": cse.ResourceMicroserviceInstance(),
 
+			"flexibleengine_dcs_backup":   dcs.ResourceDcsBackup(),
+			"flexibleengine_dcs_instance": dcs.ResourceDcsInstance(),
```
Comment (chengxiangdong) [inline flexibleengine/provider.go:504]:
```suggestion
			"flexibleengine_dcs_backup_v1":   dcs.ResourceDcsBackup(),
			"flexibleengine_dcs_instance_v1": dcs.ResourceDcsInstance(),
```

**Comment by Zhukun-Huawei** [comment]
> The commit message "tmp" lacks clear meaning, pls refine it.

i have revised the name of this pr

**Comment by chengxiangdong** [comment]
/retitle feat(dcs): import DCS resources, datasources and docs

File: docs/data-sources/dcs_flavors_v1.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
+---
+subcategory: "Distributed Cache Service (DCS)"
+---
+
+# flexibleengine_dcs_flavors_v1_v1
```
Comment (chengxiangdong) [inline docs/data-sources/dcs_flavors_v1.md:5]:
```suggestion
# flexibleengine_dcs_flavors_v1
```
a nit

File: flexibleengine/acceptance/common.go [hunk 8]
```diff
@@ -60,3 +60,62 @@ data "flexibleengine_images_image" "test" {
...
+  cache_mode     = "cluster"
+  capacity       = 8
+  replica_count  = 2
+}
+
+resource "flexibleengine_dcs_instance_v1" "instance_1" {
+  name              = "%s"
+  engine            = "Redis"
+  engine_version    = "5.0"
+  password          = "Huawei_test"
```
Comment (chengxiangdong) [inline flexibleengine/acceptance/common.go:107]:
```suggestion
  password          = "FlexibleEngine_test"
```

File: flexibleengine/provider.go [hunk 9]
```diff
@@ -496,6 +500,9 @@ func Provider() *schema.Provider {
 			"flexibleengine_cse_microservice_engine":   cse.ResourceMicroserviceEngine(),
 			"flexibleengine_cse_microservice_instance": cse.ResourceMicroserviceInstance(),
 
+			"flexibleengine_dcs_backup_v1": dcs.ResourceDcsBackup(),
+			"flexibleengine_dcs_instance":  dcs.ResourceDcsInstance(),
```
Comment (chengxiangdong) [inline flexibleengine/provider.go:504]: `flexibleengine_dcs_instance_v1` already exists.
Please remove this line.

**Review by chengxiangdong — CHANGES_REQUESTED** [review]

**Pull request merged**